### PR TITLE
Various small energy related fixes and cleanups

### PR DESCRIPTION
**Pull request by khenderick**
* Move maintenance communicator to more correct location
* Fixed logging received data on error
* Fetch phase data for P1C
* Moved certain exceptions to better location
* Enhanced some logging
* Implement power cycle bus for the Core
* Harden power.db migration code
* Actually stop the ventilation controller

### CODE REVIEW

File: src/gateway/hal/master_controller_core.py [hunk 1]
```diff
@@ -1067,7 +1067,21 @@ def _default_if_255(value, default):
     # Generic
 
     def power_cycle_bus(self):
-        raise NotImplementedError()
+        # TODO: Replace by cycle instruction as soon as it's available in the firmware
+        try:
+            logger.warning('Powering down RS485 bus...')
+            self._master_communicator.do_basic_action(BasicAction(action_type=253,
```
Comment (Shinigami-shinto) [inline src/gateway/hal/master_controller_core.py:1073]: BA 253 isn't listed on https://wiki.openmotics.com/index.php/Action_Types

File: src/gateway/maintenance_controller.py [hunk 2]
```diff
@@ -37,10 +41,11 @@ class MaintenanceController(object):
 
     @Inject
     def __init__(self, maintenance_communicator=INJECTED, ssl_private_key=INJECTED, ssl_certificate=INJECTED):
+        # type: (MaintenanceCommunicator, str, str) -> None
         """
         :type maintenance_communicator: gateway.maintenance_communicator.MaintenanceCommunicator
         """
-        self._consumers = {}
+        self._consumers = {}  # type: Dict[int, Optional[Callable[[str], Any]]]
```
Comment (Shinigami-shinto) [inline src/gateway/maintenance_controller.py:48]: assuming this type hint is correct

**Review by Shinigami-shinto — COMMENTED** [review]
Nothing too complex, wel bijgeleerd over "old" en "new style classes"

File: src/gateway/hal/master_controller_core.py (hunk 1 above)
Comment (khenderick) [reply src/gateway/hal/master_controller_core.py:1073]: Correct, it's in the Brain+ part, which is documented on https://wiki.openmotics.com/index.php/Action_Types_AIO

File: src/gateway/maintenance_controller.py (hunk 2 above)
Comment (khenderick) [reply src/gateway/maintenance_controller.py:48]: It should be, yes.

**Review by Shinigami-shinto — APPROVED** [review]

**Pull request merged**